frontend/src/components/ActionLink/ActionLinkwithClick.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import styled from 'react-emotion';
+import { Link } from 'react-router-dom';
+import { logAction } from '../../utils';
+
+const ActionLink = styled(Link)`
+  display: block;
+  text-decoration: none;
+  color: #b8b8b8;
+  font-weight: bold;
+  letter-spacing: 1.1px;
+  padding: 10px 10px;
+  border: 1px solid transparent;
+
+  &:hover {
+    color: ${({ secondary }) => (secondary ? '#eb4934' : '#3d70b2')};
+    border: 1px solid ${({ secondary }) => (secondary ? '#eb4934' : '#3d70b2')};
+  }
+  &:active {
+    color: ${({ secondary }) => (secondary ? '#eb4934' : '#3d70b2')};
+    border: 1px solid
+      ${({ secondary }) => (secondary ? '#eb4934' : 'rgba(24, 104, 184, 0.1)')};
+    background-color: ${({ secondary }) =>
+      secondary ? 'rgba(235, 72, 51, 0.1)' : 'rgba(24, 104, 184, 0.1)'};
+  }
+`;
+
+function ActionLinkwithClick({
+  data,
+  title,
+  markThreadAsDeleted,
+  markThreadAsInactive,
+  threadId,
+  remove,
+  ...restProps
```

Comment:
It would be good to add some prop typing here to ensure we are getting the correct value types into our component. We use `flow` for this. https://flow.org/en/docs/types/